### PR TITLE
Mobile view

### PR DESCRIPTION
**Pull request by aasalazar23**
Improved signup side of the site to fit mobile, tablet, and desktop sizes. 

On mobile screens, we're using a details element to hide the large filter list. In the future, we'll replace it with something more robust. 

Screen Shot 2021-06-04 at 15 58 02 PM
Screen Shot 2021-06-04 at 15 57 51 PM

### CODE REVIEW

File: src/components/all.scss [hunk 1]
```diff
@@ -480,6 +480,31 @@ img {
...
+  flex-shrink: 0;
+  line-height: 1.25;
+  text-decoration: none;
+  white-space: nowrap;
+  padding: 0.5rem 0.75rem;
+  margin: 1rem 0;
+
+  summary {
+    font-size: 1.25rem;
+    list-style-image: url(https://s3-us-west-2.amazonaws.com/s.cdpn.io/3/right-arrow.svg)
```
Comment (danielpowell4) [inline src/components/all.scss:504]: this is a code smell x 50

Comment (danielpowell4) [reply src/components/all.scss:504]: it it's in SVG you are able to inline it

File: src/components/all.scss [hunk 2]
```diff
@@ -480,6 +480,31 @@ img {
   border: 2px solid var(--accent);
 }
 
+.custom-details {
+  background-color: var(--accent);
+  border-color: var(--accent);
+  color: var(--text-primary);
```
Comment (danielpowell4) [inline src/components/all.scss:486]: is this color != to the default?

if no, why set it

File: src/components/all.scss [hunk 3]
```diff
@@ -480,6 +480,31 @@ img {
   border: 2px solid var(--accent);
 }
 
+.custom-details {
```
Comment (danielpowell4) [inline src/components/all.scss:483]: this name is far from descriptive and/or helpful to devs

File: src/components/ClassCards.scss [hunk 4]
```diff
@@ -85,5 +86,15 @@
...
     }
+
+    @media (min-width: 576px) {
+      // two column on landscape phone
+      grid-column-end: span 6;
+    }
+
+    // Medium devices (tablets, 768px and up)
+    @media (min-width: 768px) {
+      grid-column-end: span 6;
```
Comment (danielpowell4) [inline src/components/ClassCards.scss:97]: these are the same.

99% sure 2nd should be 4

File: src/pages/classes.scss [hunk 5]
```diff
@@ -1,20 +1,36 @@
...
   &__hero {
-    padding: 2rem 1rem;
     max-width: 1170px;
     margin: auto;
     width: 100%;
     color: var(--text-primary);
 
     &__title {
       font-weight: bold;
+      margin-bottom: 0.5rem;
```
Comment (danielpowell4) [inline src/pages/classes.scss:11]: I say embrace the space!

There is such thing as too many things in a few hundred pixels

File: src/utils/helpers.js [hunk 6]
```diff
@@ -21,7 +21,12 @@ export const groupBy = (arr, key) => {
...
 };
 
-export const isMobile = _ =>
-  /Android|webOS|iPhone|iPod|BlackBerry|IEMobile|Opera Mini/i.test(
-    navigator.userAgent
-  );
+export const isMobile = _ => {
+  if (typeof navigator !== "undefined") {
+    return /Android|webOS|iPhone|iPod|BlackBerry|IEMobile|Opera Mini/i.test(
+      navigator?.userAgent
```
Comment (danielpowell4) [inline src/utils/helpers.js:27]: A suggestion that you don't have to take!

Seems like you could instead do:

```
{
  if (!navigator?.userAgent) return false; // for build step
  
  return /Android|webOS|iPhone|iPod|BlackBerry|IEMobile|Opera Mini/i.test(
      navigator.userAgent 
  )
}
```

Background:

Sniffing for specific features in an API to test browser support (or lack thereof) is a normal way to include a polyfill (or progressive enhancement) or not

File: src/components/all.scss (hunk 1 above)
Comment (aasalazar23) [reply src/components/all.scss:504]: Forgot to remove this, thanks

File: src/utils/helpers.js (hunk 6 above)
Comment (aasalazar23) [reply src/utils/helpers.js:27]: I tried this exact syntax, but it was throwing an error that `navigator is not defined` Don't totally understand why as I thought `?.` would work.

**Pull request merged**